catalog/catalog-app/src/main/resources/features.xml

```diff
@@ -39,15 +39,15 @@
 
     <feature name="catalog-core" install="manual" version="${project.version}"
              description="Catalog Core feature containing the API, third party bundles necessary to run ddf-core.">
-        <feature>catalog-core-api</feature>
+        <feature prerequisite="true">catalog-core-api</feature>
```

Comment:
Yeah Karaf does a great job of not documenting these things.  You are correct. by specifying a feature as a prereq, it must be installed, prior to installing the parent feature.  We actually sort of did this before by the order in which we listed our features. That order ensured things like API bundles were installed first. In Karaf 4, they provision differently (hit me up if you want a detailed description of the change) and the order doesn't ensure one feature will start before another (which is good) and yes we do require it.  It does appear to slow down feature installation a bit (I don't have any formal metrics), but I think there are things we can do speed things up.